### PR TITLE
new lint: `file_names`

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

**Comment by pq** [comment]
Good question @bwilkerson .  I'm guessing @a14n, you ran this on `flutter`?

Regardless I'll be sure and put a BREAKING CHANGE note in the changelog.

**Comment by pq** [comment]
Ok, hmmm.  The more i think on this, the more I think we really want to break this into a separate lint.   
  I could imagine wanting to enforce one but not the other.  Am I being crazy?

What about adding one called `file_names`?

**Comment by pq** [comment]
Let's split it.  They can both point back to that piece of advice.

Thanks!

File: lib/src/rules/file_names.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import 'package:linter/src/utils.dart';
+
+const _desc = r'Name source files using `lowercase_with_underscores`.';
+
+const _details = r'''
+
+**DO** name source files using `lowercase_with_underscores`.
+
+Some file systems are not case-sensitive, so many projects require filenames
+to be all lowercase. Using a separate character allows names to still be
```
Comment (pq) [inline lib/src/rules/file_names.dart]: separate => separator?

File: lib/src/rules/library_names.dart [hunk 2]
```diff
@@ -7,12 +7,11 @@ import 'package:analyzer/dart/ast/visitor.dart';
...
-    r'Name libraries and source files using `lowercase_with_underscores`.';
+const _desc = r'Name libraries using `lowercase_with_underscores`.';
 
 const _details = r'''
 
-**DO** name libraries and source files using `lowercase_with_underscores`.
+**DO** name libraries using `lowercase_with_underscores`.
 
 Some file systems are not case-sensitive, so many projects require filenames
 to be all lowercase.  Using a separate character allows names to still be
```
Comment (pq) [inline lib/src/rules/library_names.dart]: I wonder if we should call out `file_names` as a related lint?

File: lib/src/rules/file_names.dart (hunk 1 above)
Comment (a14n) [reply lib/src/rules/file_names.dart]: I updated the sentence with the content of https://www.dartlang.org/guides/language/effective-dart/style#do-name-libraries-and-source-files-using-lowercase_with_underscores

File: lib/src/rules/library_names.dart (hunk 2 above)
Comment (a14n) [reply lib/src/rules/library_names.dart]: done

**Comment by pq** [comment]
Awesome.  Thank you!

**Pull request merged**